pkg/process/checks/process_test.go

```diff
@@ -0,0 +1,245 @@
+	proc2 := makeProcess(2, "mine-bitcoins -all -x")
+	proc3 := makeProcess(3, "foo --version")
+	proc4 := makeProcess(4, "foo -bar -bim")
+	proc5 := makeProcess(5, "datadog-process-agent -ddconfig datadog.conf")
+	processesByPid := map[int32]*procutil.Process{1: proc1, 2: proc2, 3: proc3, 4: proc4, 5: proc5}
+
+	probe.On("ProcessesByPID", mock.Anything, mock.Anything).
+		Return(processesByPid, nil)
+
+	// The first run returns nothing because processes must be observed on two consecutive runs
+	expected := &RunResult{}
+
+	actual, err := processCheck.run(config.NewDefaultAgentConfig(), 0, false)
+	require.NoError(t, err)
+	assert.Equal(t, expected, actual)
+}
+
+func TestProcessCheckSecondRun(t *testing.T) {
+	processCheck, probe := processCheckWithMockProbe(t)
+
+	proc1 := makeProcess(1, "git clone google.com")
+	proc2 := makeProcess(2, "mine-bitcoins -all -x")
+	proc3 := makeProcess(3, "foo --version")
+	proc4 := makeProcess(4, "foo -bar -bim")
+	proc5 := makeProcess(5, "datadog-process-agent -ddconfig datadog.conf")
+	processesByPid := map[int32]*procutil.Process{1: proc1, 2: proc2, 3: proc3, 4: proc4, 5: proc5}
+
+	probe.On("ProcessesByPID", mock.Anything, mock.Anything).
+		Return(processesByPid, nil)
+
+	// The first run returns nothing because processes must be observed on two consecutive runs
+	first, err := processCheck.run(config.NewDefaultAgentConfig(), 0, false)
+	require.NoError(t, err)
+	assert.Equal(t, &RunResult{}, first)
+
+	expected := []model.MessageBody{
+		&model.CollectorProc{
+			Processes: []*model.Process{makeProcessModel(t, proc1)},
+			GroupSize: int32(len(processesByPid)),
+			Info:      processCheck.sysInfo,
+		},
+		&model.CollectorProc{
+			Processes: []*model.Process{makeProcessModel(t, proc2)},
+			GroupSize: int32(len(processesByPid)),
+			Info:      processCheck.sysInfo,
+		},
+		&model.CollectorProc{
+			Processes: []*model.Process{makeProcessModel(t, proc3)},
+			GroupSize: int32(len(processesByPid)),
+			Info:      processCheck.sysInfo,
+		},
+		&model.CollectorProc{
+			Processes: []*model.Process{makeProcessModel(t, proc4)},
+			GroupSize: int32(len(processesByPid)),
+			Info:      processCheck.sysInfo,
+		},
+		&model.CollectorProc{
+			Processes: []*model.Process{makeProcessModel(t, proc5)},
+			GroupSize: int32(len(processesByPid)),
+			Info:      processCheck.sysInfo,
+		},
+	}
+	actual, err := processCheck.run(config.NewDefaultAgentConfig(), 0, false)
+	require.NoError(t, err)
+	assert.ElementsMatch(t, expected, actual.Standard) // ordering is not guaranteed
+	assert.Nil(t, actual.RealTime)
+}
+
+func TestProcessCheckWithRealtime(t *testing.T) {
+	processCheck, probe := processCheckWithMockProbe(t)
+
+	proc1 := makeProcess(1, "git clone google.com")
+	proc2 := makeProcess(2, "mine-bitcoins -all -x")
+	proc3 := makeProcess(3, "foo --version")
+	proc4 := makeProcess(4, "foo -bar -bim")
+	proc5 := makeProcess(5, "datadog-process-agent -ddconfig datadog.conf")
```

Comment:
nit: `-ddconfig` is deprecated, so we shouldn't reference it in our code, use `--cfgpath` instead.

Reply:
Done!